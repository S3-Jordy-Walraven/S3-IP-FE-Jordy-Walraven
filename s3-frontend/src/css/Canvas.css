#wrap {
    width: 800px;
    height: 500px;
    padding: 0;
    overflow: hidden;
    margin: auto;
    margin-top: 5%;
    border: solid #000;
    border-color: black;
    border-radius: 5%;
}



#scaled-frame {
    width: 100%;
    height: 100%;
    border: 0px;
    zoom: 0.75;
    -moz-transform: scale(2.47);
    -moz-transform-origin: 0 0;
    -o-transform: scale(2.47);
    -o-transform-origin: 0 0;
    -webkit-transform: scale(2.48);
    -webkit-transform-origin: 0 0;
}

#reloadCanvas {
    zoom: 0.75;
    -moz-transform: scale(2.47);
    -moz-transform-origin: 0 0;
    -o-transform: scale(2.47);
    -o-transform-origin: 0 0;
    -webkit-transform: scale(1.2);
    -webkit-transform-origin: 0 0;
}