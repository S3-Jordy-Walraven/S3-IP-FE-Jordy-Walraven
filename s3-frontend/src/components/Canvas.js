import React, { useEffect } from "react";
import CanvasService from "../services/CanvasService";
import "../css/Canvas.css"

const Canvas = (props) => {
  var cService = new CanvasService();
  function GetHtml(content) {
    if (content === false) {
      cService.getHtmlText(props.fileInput, GetHtml);
    } else {
      console.log(content)
      var head = content.split("</head>")[0].replace(`\\`,"");
      var metaProp = head.split("property")
      var vars = [];
      var metaDefaults = head.split("default");
      var defaultValues = [];
      var types = head.split("type");
      var typeValues = [];
      for (var i = 0; i < metaProp.length; i++) {
        if (i !== 0 ) 
          vars.push(metaProp[i].split(`="`)[1].split('"')[0]);
      }
      console.log(metaDefaults)
      for (let index = 0; index < metaDefaults.length; index++) {
        if (index !== 0)
          defaultValues.push(metaDefaults[index].split(`="`)[1].split('"')[0]);
      }
      for (let index = 0; index < types.length; index++) {
        if (index !== 0)
          typeValues.push(types[index].split(`="`)[1].split('"')[0]);
      }
      var num = content.split("<script>")[0].length + 8;
      var script = content.substring(0, num);
      vars.forEach((element, i) => {
        if(typeValues[i] === "number"){
          script += `var ${element} =  ${defaultValues[i]}\n `
        } else{
          script += `var ${element} =  "${defaultValues[i]}"\n `
        }
  
      });
      var html = script + content.substring(num, content.length);

      props.htmlStringCallback(html);
    }
  }

  useEffect(() => {
    console.log(props.fileInput)
    if (props.fileInput)
      GetHtml(false);
  }, [props.fileInput]);

  useEffect(() => {

  }, [props.finalString]);



  return (
    <div id="wrap">
      <iframe title="canvas" id="scaled-frame" srcDoc={props.finalString}  sandbox="allow-scripts " width="100%" height={100} >
      </iframe>
    </div>
  );
};

export default Canvas;
