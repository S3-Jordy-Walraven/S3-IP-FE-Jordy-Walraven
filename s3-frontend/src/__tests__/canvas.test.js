import { render, screen, cleanup, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import Canvas from "../components/Canvas";
import userEvent from "@testing-library/user-event";
import { useState } from "react";
import Html from "./files/Hyperspace.html"

afterEach(cleanup);

test("should render canvas", () => {
  render(<Canvas />);
  const canvasElement = screen.getByTestId("canvas-1");
  expect(canvasElement).toBeInTheDocument();
});

test("render effect", async () => {
  let finalString = "a";

  async function StringCallback(html) {
    finalString = html;
  }

  render(
    <Canvas
      fileInput={
        new File(["<script>console.log('hello')</script>"], "test.html")
      }
      htmlStringCallback={StringCallback}
      finalString={finalString}
    />
  );
  await waitFor(() => {
    expect(finalString).toBe("<script>console.log('hello')</script>");
  });
});

test("matches snapshot", () => {
  const tree = renderer.create(<Canvas />).toJSON();
  expect(tree).toMatchSnapshot();
});
