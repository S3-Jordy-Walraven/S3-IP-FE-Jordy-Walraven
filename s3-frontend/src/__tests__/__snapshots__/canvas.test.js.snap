// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="canvas-1"
  id="wrap"
>
  <iframe
    height={100}
    id="scaled-frame"
    sandbox="allow-scripts "
    title="canvas"
    width="100%"
  />
</div>
`;
